src/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -133,17 +136,29 @@
  */
 final class SearchFilter extends AbstractFilter implements SearchFilterInterface
 {
+    use ResourceClassInfoTrait;
     use SearchFilterTrait;
 
     public const DOCTRINE_INTEGER_TYPE = Types::INTEGER;
 
-    public function __construct(ManagerRegistry $managerRegistry, IriConverterInterface $iriConverter, PropertyAccessorInterface $propertyAccessor = null, LoggerInterface $logger = null, array $properties = null, IdentifiersExtractorInterface $identifiersExtractor = null, NameConverterInterface $nameConverter = null)
-    {
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        IriConverterInterface $iriConverter,
+        PropertyAccessorInterface $propertyAccessor = null,
+        LoggerInterface $logger = null,
+        array $properties = null,
+        IdentifiersExtractorInterface $identifiersExtractor = null,
+        NameConverterInterface $nameConverter = null,
+        ResourceMetadataCollectionFactoryInterface $resourceMetadataFactory = null,
+        ResourceClassResolverInterface $resourceClassResolver = null
+    ) {
         parent::__construct($managerRegistry, $logger, $properties, $nameConverter);
 
         $this->iriConverter = $iriConverter;
         $this->identifiersExtractor = $identifiersExtractor;
         $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->resourceClassResolver = $resourceClassResolver;
```

Comment:
lots of new dependencies I wish we wouldn't have...

Reply:
yes, that's mostly why i didn't include this part in the previous PR. And that's kind of the point I was making in https://github.com/api-platform/core/pull/5687 - some of the metadata is not easy to get to at times.